src/tainting/Dataflow_tainting.ml

```diff
@@ -1435,8 +1435,17 @@ let check_function_call_arguments env args =
  * report the finding too (by side effect). *)
 let check_tainted_instr env instr : Taints.t * Lval_env.t =
   let check_expr env = check_tainted_expr env in
-  let check_instr = function
-    | Assign (_, e) -> check_expr env e
+  let rec check_instr = function
+    | Assign (_, e) -> (
+        match e.e with
+        (* Considering that Ruby permits function calls without parameters or
+         * parentheses, we are checking whether the identifier has a function
+         * signature specifically for Ruby. *)
+        | Fetch _
```

Comment:
Not very fond of doing it here. Could you try doing this hack in AST_to_IL? It would be almost the same thing, but then you generate a `Call` instead of an `Assign` when these conditions are met.  Ideally we check the type of the lval (for Pro) or the existence of a funcdef for this lval, but you can leave that as a TODO for now. Ideally this is fixed already in the Generic AST indeed, but it's OK having a temporary hack in AST_to_IL.

Reply:
Fixed.